janitor/timeseries.py

```diff
@@ -176,3 +180,273 @@ def sort_timestamps_monotonically(
+            )
+        )
+
+        # Applies specific criteria to certain dataframe columns
+        # Applies default criteria to columns not specifically listed
+        # Appends a flag column for each column in the dataframe
+        df = (
+            pd.DataFrame(...)
+            .flag_jumps(
+                scale=dict(col1="absolute"),
+                direction=dict(col2="increasing"),
+            )
+        )
+
+        # Applies specific criteria to certain dataframe columns
+        # Applies default criteria to columns not specifically listed
+        # Appends a flag column for only those columns found in
+        #   specified criteria
+        df = (
+            pd.DataFrame(...)
+            .flag_jumps(
+                scale=dict(col1="absolute"),
+                threshold=dict(col2=1),
+                strict=True,
+            )
+        )
+
+    :param df: Dataframe which needs to be flagged for changes between
+        consecutive rows above a certain threshold.
+    :param scale: Type of scaling approach to use.
+        Acceptable arguments are:
+            1. absolute (consider the difference between rows).
+            2. percentage (consider the percentage change between rows).
+        Defaults to percentage.
+    :param direction: Type of method used to handle the sign change when
+        comparing consecutive rows.
+        Acceptable arguments are:
+            1. increasing (only consider rows that are increasing in value).
+            2. decreasing (only consider rows that are decreasing in value).
+            3. any (consider rows that are either increasing or decreasing;
+                sign is ignored).
+        Defaults to any.
+    :param threshold: The value to check if consecutive row comparisons
+        exceed. Always uses a greater than comparison. Must be >= 0.0.
+        Defaults to 0.0
+    :param strict: flag to enable/disable appending of a flag column for
+        each column in the provided dataframe.
+        If set to True, will only append a flag column for those columns
+            found in at least one of the input dictionaries.
+        If set to False, will append a flag column for each column found
+             in the provided dataframe. If criteria is not specified,
+             the defaults for each criteria is used.
+        Defaults to False.
+    :returns: Dataframe that has flag jump columns.
+    :raises: JanitorError if ``strict=True`` and at least one of
+        ``scale``, ``direction``, or ``threshold`` inputs is not a
+        dictionary.
+    :raises: JanitorError if ``scale`` is not one of
+        ``["absolute", "percentage"]``.
+    :raises: JanitorError if ``direction`` is not one of
+        ``["increasing", "decreasing", "any"]``.
+    :raises: JanitorError if ``threshold`` is less than 0.0.
+    """
+    df = df.copy()
+
+    if strict:
+        if (
+            any(isinstance(arg, dict) for arg in (scale, direction, threshold))
+            is False
```

Comment:
really not sure about this one, using the `is` here; what if we affixed `not` in front of any, removing the need for `is False`, or using `all` ?

Reply:
I used `is False` as it seems a little more intuitive to me in terms of readability... I can read this line and it makes sense saying it out loud... "If any <of these> is False..."

Perhaps the use of `not` is more Pythonic, but I was going for readability. "If not any()" or "If not all()" is harder to interpret (to me) that "If any is False".

Happy to change it goes against best practices.

Reply:
This condition looks a bit more complex, so explicit readability is a good idea here. Let's stick with what you have, @loganthomas.